### PR TITLE
fix: added telemetry support to node runtime and adopted missing middlewares

### LINKED ISSUE #5808: Node runtime unable to send trace and customEvent to application insights
## Describe the bug

Logs not sent to application insight after providing instrumentation key

## Version

Version: 1.3.0
NodeJS: 12.13.0

## OS

- [x] macOS
- [ ] Windows
- [ ] Ubuntu

## To Reproduce

Create new bot composer project
Add 'Emit a telemetry track event'
Add instrumentation key in advanced settings view
Use custom runtime with location 'runtime' and start command 'node ./lib/webapp.js'
Run dialog to hit emit event

## Expected behavior

Expected log to appear in application insights

## Additional context

Noticed that the .NET version of runtime has code where application insights is added to dialog manager, similar code is missing in the Node runtime.
Was able to send telemetry by adding useTelemetry() inside loadRootDialog()

### PR DESCRIPTION
**Pull request by chon219**
## Description

Added telemetry support to node runtime and adopted missing middlewares.

## Task Item

Fix #5808 
Added following middlewares to node runtime as well:
* TranscriptLoggerMiddleware
* InspectionMiddleware
* ShowTypingMiddleware

### CODE REVIEW

**Comment by a-b-r-o-w-n** [comment]
@benbrown should this be included in the upcoming release? Can you be the owner?

File: runtime/node/src/webapp.ts [hunk 1]
```diff
@@ -44,6 +54,9 @@ configureSkillEndpoint(server, adapter, bot, skillConversationIdFactory);
 // Configure manifests endpoint.
 configureManifestsEndpoint(server);
 
+// Configure telemetry client, initializers and middleware.
+configureTelemetry(server, adapter, telemetryClient);
```
[the remark this answers is not in the document]
Comment (chon219) [reply runtime/node/src/webapp.ts]: `configureTelemetry` should be shared by both webapp and functions, so I've moved the `server.use(ApplicationInsightsWebserverMiddleware)` part into webapp.ts and let functions use `configureTelemetry` as well.

File: runtime/node/src/shared/helpers.ts [hunk 2]
```diff
@@ -113,14 +126,40 @@ export const getRootDialog = (folderPath: string): string => {
...
+  conversationState: ConversationState
+): BotFrameworkAdapter => {
   const settings = getSettings();
   const adapterSettings: Partial<BotFrameworkAdapterSettings> = {
     appId: settings.MicrosoftAppId,
     appPassword: settings.MicrosoftAppPassword,
   };
   const adapter = new BotFrameworkAdapter(adapterSettings);
   useBotState(adapter, userState, conversationState);
+
```
[the remark this answers is not in the document]
Comment (chon219) [reply runtime/node/src/shared/helpers.ts:141]: No, they are some minor missing parts in node runtime and I think they are also required. I should have documented them in the PR.

**Comment by carlosscastro** [comment]
@benbrown @luhan2017 @chon219 could we get more traction on this PR?

File: runtime/node/package.json [hunk 3]
```diff
@@ -19,6 +19,8 @@
     "adaptive-expressions": "4.12.0-rc3",
     "botbuilder": "4.12.0-rc3",
     "botbuilder-ai": "4.12.0-rc3",
+    "botbuilder-applicationinsights": "4.12.0-rc3",
```
Comment (luhan2017) [inline runtime/node/package.json:22]: let's update this to the released version instead of rc.

**Review by luhan2017 — APPROVED** [review]

File: runtime/node/package.json (hunk 3 above)
Comment (chon219) [reply runtime/node/package.json]: done.

**Review by luhan2017 — APPROVED** [review]

**Pull request merged**